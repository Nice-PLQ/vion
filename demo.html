<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport"
    content="width=device-width, initial-scale=1, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no,minimal-ui, viewport-fit=cover">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="icon" href="./assets/favicon.dfb64e57.ico">
  <title>vion - 基于QQ Design设计语言的组件库</title>
  <script src="https://cdn.jsdelivr.net/npm/@vant/touch-emulator"></script>
  <script type="module" crossorigin src="./assets/demo.3270c561.js"></script>
  <link rel="modulepreload" href="./assets/index.466186a3.js">
  <link rel="stylesheet" href="./assets/index.e569a14b.css">
  <link rel="stylesheet" href="./assets/demo.16f455e0.css">
</head>

<body ontouchstart>
  <div id="popup-el"></div>
  <div id="app"></div>
  <script src="https://pub.idqqimg.com/qqmobile/qqapi.js?_bid=152"></script>
  
</body>

</html>
