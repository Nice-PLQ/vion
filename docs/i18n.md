# 国际化

Vion 默认采用中文语言，如果需要切换其他语言，请参考如下

### 1、切换语言

Vion 通过 Locale 组件实现多语言支持，使用`Locale.use`方法切换语言环境，比如切换到英文环境

```js
// 按需引入
import { createApp } from "vue";
import { ActionSheet, Locale } from "vion";
import messages from "vion/es/locale/lang/en-US";

Locale.use("en-US", messages);

createApp()
  .use(ActionSheet);
  .use(Locale);

```

或

```js
// 全量引入Vion
import { createApp } from "vue";
import Vion from "vion";
import messages from "vion/es/locale/lang/en-US";

vion.Locale.use("en-US", messages);

createApp().use(Vion);
```

### 2、语言包

目前 vion 内置了以下几种语言

- 简体中文（zh-CN）
- 繁体中文（zh-Hant）
- 英语（en-US）
- 日语（ja-JP）

如果不能满足业务场景，只需要按照[语言包格式](https://github.com/Nice-PLQ/vion/blob/master/src/locale/lang/zh-CN.js)自定义语言，然后使用`Locale`组件加载即可，如：

```js
import messages from "./lang/fr-FR"; // 自定义的【法语】语言包
Locale.use("fr-FR", messages);
```

### 3、引入多个语言包

```js
import frMessages from "./lang/fr-FR"; // 自定义的【法语】语言包
import trMessages from "./lang/tr-TR"; // 自定义的【土耳其语】语言包

Locale.add("tr-TR", trMessages); // 将语言包加入到集合中
Locale.use("fr-FR", frMessages); // 默认使用法语作为首选语言
```
