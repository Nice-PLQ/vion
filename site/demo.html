<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport"
    content="width=device-width, initial-scale=1, maximum-scale=1.0, minimum-scale=1.0, user-scalable=no,minimal-ui, viewport-fit=cover">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="icon" href="/favicon.ico">
  <title>vion - 基于QQ Design设计语言的组件库</title>
  <script src="https://cdn.jsdelivr.net/npm/@vant/touch-emulator"></script>
</head>

<body ontouchstart>
  <div id="popup-el"></div>
  <div id="app"></div>
  <script src="https://pub.idqqimg.com/qqmobile/qqapi.js?_bid=152"></script>
  <script type="module" src="./demo/index.js"></script>
</body>

</html>
