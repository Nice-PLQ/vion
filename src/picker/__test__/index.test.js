import { mount } from '@vue/test-utils';
import Picker from '..';
import { years, months, days } from '../demo/data';
import { triggerDrag, next } from '../../../test';
import i18n from '../../locale';
import VPopup from '../../popup';

describe('Picker', () => {
  test('列的数量', async () => {
    const wrapper = mount(Picker, {
      props: {
        columns: [years, months, days],
        teleportDisabled: true,
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    expect(wrapper.findAll('.v-picker-list')).toHaveLength(3);
    expect(wrapper.find('.v-picker').element).toMatchSnapshot();

    wrapper.setProps({ columns: [[{ label: '2019年' }, ...years], months] });
    await next();
    expect(wrapper.findAll('.v-picker-list')).toHaveLength(2);
  });

  test('列显示的个数', async () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years,
        showItemCount: 7
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    await next();
    expect(wrapper.find('.v-picker__content').element.style.height).toBe(`${7 * 46}px`);
  });

  test('隐藏取消按钮', () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years,
        showCancel: false
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        }
      }
    });
    expect(wrapper.findAll('.v-picker__btn')).toHaveLength(1);
    expect(wrapper.findAll('.v-picker__btn')[0].text()).toBe('确定');
  });

  test('默认选中', async () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    const selectedValue = { label: '2025年' };
    wrapper.setProps({ selectedValue });
    await next();
    expect(wrapper.vm.getValues()).toEqual(selectedValue);
  });

  test('拖动列', async () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years,
        immediateChange: true
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    const el = wrapper.find('.v-picker-column');
    triggerDrag(el, 0, -500);
    el.trigger('transitionend');
    expect(wrapper.emitted('change')[0][1]).toEqual({ label: '2029年' });
  });

  test('change && cancel && ok事件的触发', async () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years,
        visible: true
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    wrapper.findAll('.v-picker-column__item')[1].trigger('click');
    await next();
    expect(wrapper.emitted('change')).toBeTruthy();

    const btn = wrapper.findAll('.v-picker__btn');
    btn[0].trigger('click');
    expect(wrapper.emitted('cancel')).toBeTruthy();
    expect(wrapper.emitted('close')).toBeTruthy();

    wrapper.setProps({ visible: true });
    await next();
    btn[1].trigger('click');
    expect(wrapper.emitted('ok')).toBeTruthy();
  });

  test('setValues方法', async () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    // 单列
    const val = { label: '2025年' };
    wrapper.vm.setValues(val);
    await next();
    expect(wrapper.vm.getValues()).toEqual(val);
    expect(wrapper.vm.getColumnSelectValue(0)).toEqual(val);

    // 多列
    const val2 = [{ label: '2025年' }, { label: '2月' }];
    wrapper.setProps({ columns: [years, months] });
    await next();
    wrapper.vm.setValues(val2);
    expect(wrapper.vm.getValues()).toEqual(val2);
  });

  test('setColumnValues方法', async () => {
    const wrapper = mount(Picker, {
      props: {
        teleportDisabled: true,
        columns: years
      },
      global: {
        config: {
          globalProperties: {
            $locale: i18n,
          }
        },
        components: {
          VPopup
        },
      }
    });
    wrapper.vm.setColumnValues(0, months);
    await next();
    expect(wrapper.findAll('.v-picker-column__item')).toHaveLength(12);
  });
});
