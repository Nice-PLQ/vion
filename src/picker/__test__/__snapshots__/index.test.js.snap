// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picker 列的数量 1`] = `
<div
  aria-modal="true"
  class="v-picker"
>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="true"
  >
    <div
      aria-label="关闭"
      class="v-popup__mask"
      role="button"
      style="display: none;"
    />
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="fade"
    persisted="false"
  >
    <!--v-if-->
  </transition-stub>
  <transition-stub
    appear="false"
    css="true"
    name="slide-bottom"
    persisted="true"
  >
    <div
      class="v-popup__content v-popup__content_bottom"
      style="display: none;"
    >
      
      <div
        class="v-picker-box"
      >
        <div
          class="v-picker__header"
        >
          <!--v-if-->
          <button
            class="v-picker__btn"
          >
            取消
          </button>
          
          <div
            class="v-picker__title"
          />
          
          <button
            class="v-picker__btn"
          >
            确定
          </button>
        </div>
        <div
          class="v-picker__content"
          style="height: 230px;"
        >
          <div
            class="v-picker__indicator"
          />
          
          <div
            class="v-picker-list"
          >
            <ul
              class="v-picker-column"
              style="transform: translate3d(0, 0px, 0); transition: transform .8s cubic-bezier(.17,.97,.2,.98);"
            >
              
              <li
                class="v-picker-column__item"
              >
                2020年
              </li>
              <li
                class="v-picker-column__item"
              >
                2021年
              </li>
              <li
                class="v-picker-column__item"
              >
                2022年
              </li>
              <li
                class="v-picker-column__item"
              >
                2023年
              </li>
              <li
                class="v-picker-column__item"
              >
                2024年
              </li>
              <li
                class="v-picker-column__item"
              >
                2025年
              </li>
              <li
                class="v-picker-column__item"
              >
                2026年
              </li>
              <li
                class="v-picker-column__item"
              >
                2027年
              </li>
              <li
                class="v-picker-column__item"
              >
                2028年
              </li>
              <li
                class="v-picker-column__item"
              >
                2029年
              </li>
              
            </ul>
          </div>
          <div
            class="v-picker-list"
          >
            <ul
              class="v-picker-column"
              style="transform: translate3d(0, 0px, 0); transition: transform .8s cubic-bezier(.17,.97,.2,.98);"
            >
              
              <li
                class="v-picker-column__item"
              >
                1月
              </li>
              <li
                class="v-picker-column__item"
              >
                2月
              </li>
              <li
                class="v-picker-column__item"
              >
                3月
              </li>
              <li
                class="v-picker-column__item"
              >
                4月
              </li>
              <li
                class="v-picker-column__item"
              >
                5月
              </li>
              <li
                class="v-picker-column__item"
              >
                6月
              </li>
              <li
                class="v-picker-column__item"
              >
                7月
              </li>
              <li
                class="v-picker-column__item"
              >
                8月
              </li>
              <li
                class="v-picker-column__item"
              >
                9月
              </li>
              <li
                class="v-picker-column__item"
              >
                10月
              </li>
              <li
                class="v-picker-column__item"
              >
                11月
              </li>
              <li
                class="v-picker-column__item"
              >
                12月
              </li>
              
            </ul>
          </div>
          <div
            class="v-picker-list"
          >
            <ul
              class="v-picker-column"
              style="transform: translate3d(0, 0px, 0); transition: transform .8s cubic-bezier(.17,.97,.2,.98);"
            >
              
              <li
                class="v-picker-column__item"
              >
                1日
              </li>
              <li
                class="v-picker-column__item"
              >
                2日
              </li>
              <li
                class="v-picker-column__item"
              >
                3日
              </li>
              <li
                class="v-picker-column__item"
              >
                4日
              </li>
              <li
                class="v-picker-column__item"
              >
                5日
              </li>
              <li
                class="v-picker-column__item"
              >
                6日
              </li>
              <li
                class="v-picker-column__item"
              >
                7日
              </li>
              <li
                class="v-picker-column__item"
              >
                8日
              </li>
              <li
                class="v-picker-column__item"
              >
                9日
              </li>
              <li
                class="v-picker-column__item"
              >
                10日
              </li>
              <li
                class="v-picker-column__item"
              >
                11日
              </li>
              <li
                class="v-picker-column__item"
              >
                12日
              </li>
              <li
                class="v-picker-column__item"
              >
                13日
              </li>
              <li
                class="v-picker-column__item"
              >
                14日
              </li>
              <li
                class="v-picker-column__item"
              >
                15日
              </li>
              <li
                class="v-picker-column__item"
              >
                16日
              </li>
              <li
                class="v-picker-column__item"
              >
                17日
              </li>
              <li
                class="v-picker-column__item"
              >
                18日
              </li>
              <li
                class="v-picker-column__item"
              >
                19日
              </li>
              <li
                class="v-picker-column__item"
              >
                20日
              </li>
              <li
                class="v-picker-column__item"
              >
                21日
              </li>
              <li
                class="v-picker-column__item"
              >
                22日
              </li>
              <li
                class="v-picker-column__item"
              >
                23日
              </li>
              <li
                class="v-picker-column__item"
              >
                24日
              </li>
              <li
                class="v-picker-column__item"
              >
                25日
              </li>
              <li
                class="v-picker-column__item"
              >
                26日
              </li>
              <li
                class="v-picker-column__item"
              >
                27日
              </li>
              <li
                class="v-picker-column__item"
              >
                28日
              </li>
              <li
                class="v-picker-column__item"
              >
                29日
              </li>
              <li
                class="v-picker-column__item"
              >
                30日
              </li>
              
            </ul>
          </div>
          
        </div>
      </div>
      
    </div>
  </transition-stub>
</div>
`;
